tests/test_inclined_exponential.py

```diff
@@ -0,0 +1,102 @@
+import numpy as np
+import os
+import sys
+
+from galsim_test_helpers import *
+
+try:
+    import galsim
+except ImportError:
+    path, filename = os.path.split(__file__)
+    sys.path.append(os.path.abspath(os.path.join(path, "..")))
+    import galsim
+
+# set up any necessary info for tests
+# Note that changes here should match changes to test image files
+image_dir = './inclined_exponential_images'
+
+# Values here are strings, so the filenames will be sure to work (without truncating zeros)
+image_inc_angles = ("0.0", "1.3", "0.2", "0.01", "0.1", "0.78")
+image_scale_radii = ("3.0", "3.0", "3.0", "3.0", "2.0", "2.0")
+image_scale_heights = ("0.3", "0.5", "0.5", "0.5", "1.0", "0.5")
+image_pos_angles = ("0.0", "0.0", "0.0", "0.0", "-0.2", "-0.2")
+image_nx = 64
+image_ny = 64
+oversampling = 1.0
+
+@timer
+def test_inclined_exponential():
+    """Test that the inclined exponential profile matches the results from Lance Miller's code."""
+    
+    for inc_angle, scale_radius, scale_height, pos_angle in zip(image_inc_angles,image_scale_radii,
+                                                                image_scale_heights,image_pos_angles):
+        image_filename = "galaxy_"+inc_angle+"_"+scale_radius+"_"+scale_height+"_"+pos_angle+".fits"
+        image = galsim.fits.read(image_filename, image_dir)
+        
+        # Get float values for the details
+        inc_angle=float(inc_angle)
+        scale_radius=float(scale_radius)/oversampling
+        scale_height=float(scale_height)/oversampling
+        pos_angle=float(pos_angle)
+        
+        # Now make a test image
+        test_profile = galsim.InclinedExponential(inc_angle*galsim.radians,scale_radius,scale_height,
+                                                  gsparams=galsim.GSParams(maximum_fft_size=5000))
+        
+        # Rotate it by the position angle
+        test_profile = test_profile.rotate(pos_angle*galsim.radians)
+        
+        # Draw it onto an image
+        test_image = galsim.Image(image_nx,image_ny,scale=1.0)
+        test_profile.drawImage(test_image,offset=(0.5,0.5)) # Offset to match Lance's
+        
+        # Compare to the example - Due to the different fourier transforms used, some offset is expected,
+        # so we just compare in the core to two decimal places
+        
+        image_core = image.array[image_ny//2-2:image_ny//2+3, image_nx//2-2:image_nx//2+3]
+        test_image_core = test_image.array[image_ny//2-2:image_ny//2+3, image_nx//2-2:image_nx//2+3]
+        
+        ratio_core = image_core / test_image_core
+        
+        # galsim.fits.write(test_image,"test_"+image_filename,image_dir)
+        
+        np.testing.assert_array_almost_equal(ratio_core, np.mean(ratio_core)*np.ones_like(ratio_core), decimal = 2,
+                                             err_msg = "Error in comparison of inclined exponential profile to samples.",
+                                             verbose=True)
```

Comment:
Need to add some more testing aside from just the regression tests though.

1. Need to check the accuracy of the repr, pickling etc.  cf. uses of the function `do_pickle` in other test files.
2. Some tests of edge-on inclination == Pi/2.
3. Should test close, but not quite edge-on.  Probably on both sides of the magic cut-off you impose on i.
4. Should test that `i==0` is functionally equivalent to an `Exponential`.
5. Check some basic properties like that kValue at k=maxK() is below `maxk_threshold` and that less than `folding_threshold` fraction of the light falls outside of r=Pi/stepK() for a few inclination angles.
6. Sanity checks like kValue(0,0) == flux, centroid() == (0,0), total flux draw with large enough image == flux.